internal/edgedbtypes/range.go

```diff
@@ -0,0 +1,1054 @@
+// This source file is part of the EdgeDB open source project.
+//
+// Copyright 2020-present EdgeDB Inc. and the EdgeDB authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package edgedbtypes
+
+import "encoding/json"
+
+type emptyRangeJSON struct {
+	Empty bool `json:"empty"`
+}
+
+// NewRangeInt32 creates a new RangeInt32 value.
+func NewRangeInt32(
```

Comment:
How do I construct an empty range?

Reply:
pass equal upper and lower bounds where at least one of them is not inclusive

Reply:
I was being conservative on the API here. We can add `NewEmptyRangeX` or something if the current API is to obscure.